diff --git a/network.go b/network.go
index 1397de0..ac00b43 100644
--- a/network.go
+++ b/network.go
@@ -10,6 +10,7 @@ import (
 	"github.com/dotcloud/docker/utils"
 	"log"
 	"net"
+	"os/exec"
 	"strconv"
 	"sync"
 )
@@ -149,7 +150,7 @@ func CreateBridgeIface(config *DaemonConfig) error {
 	}
 	utils.Debugf("Creating bridge %s with network %s", config.BridgeIface, ifaceAddr)
 
-	if err := netlink.NetworkLinkAdd(config.BridgeIface, "bridge"); err != nil {
+	if err := exec.Command("brctl", "addbr", config.BridgeIface).Run(); err != nil {
 		return fmt.Errorf("Error creating bridge: %s", err)
 	}
 	iface, err := net.InterfaceByName(config.BridgeIface)
