diff -uNr docker-1.1.2/archive/archive.go docker-1.1.2-1/archive/archive.go
--- docker-1.1.2/archive/archive.go	2014-07-22 19:29:56.000000000 -0500
+++ docker-1.1.2-1/archive/archive.go	2014-07-25 10:47:31.042199877 -0500
@@ -18,7 +18,7 @@
 
 	"github.com/dotcloud/docker/pkg/system"
 	"github.com/dotcloud/docker/utils"
-	"github.com/dotcloud/docker/vendor/src/code.google.com/p/go/src/pkg/archive/tar"
+	"archive/tar"
 )
 
 type (
diff -uNr docker-1.1.2/archive/archive_test.go docker-1.1.2-1/archive/archive_test.go
--- docker-1.1.2/archive/archive_test.go	2014-07-22 19:29:56.000000000 -0500
+++ docker-1.1.2-1/archive/archive_test.go	2014-07-25 10:46:52.199198162 -0500
@@ -11,7 +11,7 @@
 	"testing"
 	"time"
 
-	"github.com/dotcloud/docker/vendor/src/code.google.com/p/go/src/pkg/archive/tar"
+	"archive/tar"
 )
 
 func TestCmdStreamLargeStderr(t *testing.T) {
diff -uNr docker-1.1.2/archive/changes.go docker-1.1.2-1/archive/changes.go
--- docker-1.1.2/archive/changes.go	2014-07-22 19:29:56.000000000 -0500
+++ docker-1.1.2-1/archive/changes.go	2014-07-25 10:47:19.873199418 -0500
@@ -12,7 +12,7 @@
 
 	"github.com/dotcloud/docker/pkg/system"
 	"github.com/dotcloud/docker/utils"
-	"github.com/dotcloud/docker/vendor/src/code.google.com/p/go/src/pkg/archive/tar"
+	"archive/tar"
 )
 
 type ChangeType int
diff -uNr docker-1.1.2/archive/diff.go docker-1.1.2-1/archive/diff.go
--- docker-1.1.2/archive/diff.go	2014-07-22 19:29:56.000000000 -0500
+++ docker-1.1.2-1/archive/diff.go	2014-07-25 10:47:07.541198850 -0500
@@ -9,7 +9,7 @@
 	"strings"
 	"syscall"
 
-	"github.com/dotcloud/docker/vendor/src/code.google.com/p/go/src/pkg/archive/tar"
+	"archive/tar"
 )
 
 // Linux device nodes are a bit weird due to backwards compat with 16 bit device nodes.
diff -uNr docker-1.1.2/archive/wrap.go docker-1.1.2-1/archive/wrap.go
--- docker-1.1.2/archive/wrap.go	2014-07-22 19:29:56.000000000 -0500
+++ docker-1.1.2-1/archive/wrap.go	2014-07-25 10:46:59.077198483 -0500
@@ -2,7 +2,7 @@
 
 import (
 	"bytes"
-	"github.com/dotcloud/docker/vendor/src/code.google.com/p/go/src/pkg/archive/tar"
+	"archive/tar"
 	"io/ioutil"
 )
 
diff -uNr docker-1.1.2/graph/tags_unit_test.go docker-1.1.2-1/graph/tags_unit_test.go
--- docker-1.1.2/graph/tags_unit_test.go	2014-07-22 19:29:56.000000000 -0500
+++ docker-1.1.2-1/graph/tags_unit_test.go	2014-07-25 10:47:40.758200323 -0500
@@ -6,7 +6,7 @@
 	_ "github.com/dotcloud/docker/daemon/graphdriver/vfs" // import the vfs driver so it is used in the tests
 	"github.com/dotcloud/docker/image"
 	"github.com/dotcloud/docker/utils"
-	"github.com/dotcloud/docker/vendor/src/code.google.com/p/go/src/pkg/archive/tar"
+	"archive/tar"
 	"io"
 	"os"
 	"path"
diff -uNr docker-1.1.2/integration/api_test.go docker-1.1.2-1/integration/api_test.go
--- docker-1.1.2/integration/api_test.go	2014-07-22 19:29:56.000000000 -0500
+++ docker-1.1.2-1/integration/api_test.go	2014-07-25 10:47:58.985201130 -0500
@@ -18,7 +18,7 @@
 	"github.com/dotcloud/docker/engine"
 	"github.com/dotcloud/docker/image"
 	"github.com/dotcloud/docker/runconfig"
-	"github.com/dotcloud/docker/vendor/src/code.google.com/p/go/src/pkg/archive/tar"
+	"archive/tar"
 )
 
 func TestGetImagesJSON(t *testing.T) {
diff -uNr docker-1.1.2/utils/tarsum.go docker-1.1.2-1/utils/tarsum.go
--- docker-1.1.2/utils/tarsum.go	2014-07-22 19:29:56.000000000 -0500
+++ docker-1.1.2-1/utils/tarsum.go	2014-07-25 10:49:00.034203856 -0500
@@ -5,7 +5,7 @@
 	"compress/gzip"
 	"crypto/sha256"
 	"encoding/hex"
-	"github.com/dotcloud/docker/vendor/src/code.google.com/p/go/src/pkg/archive/tar"
+	"archive/tar"
 	"hash"
 	"io"
 	"sort"
diff -uNr docker-1.1.2/utils/tarsum_test.go docker-1.1.2-1/utils/tarsum_test.go
--- docker-1.1.2/utils/tarsum_test.go	2014-07-22 19:29:56.000000000 -0500
+++ docker-1.1.2-1/utils/tarsum_test.go	2014-07-25 10:49:19.202204685 -0500
@@ -9,7 +9,7 @@
 	"os"
 	"testing"
 
-	"github.com/dotcloud/docker/vendor/src/code.google.com/p/go/src/pkg/archive/tar"
+	"archive/tar"
 )
 
 type testLayer struct {
