diff --git a/archive/archive.go b/archive/archive.go
index 2fac18e..634e9c1 100644
--- a/archive/archive.go
+++ b/archive/archive.go
@@ -8,7 +8,7 @@ import (
 	"fmt"
 	"github.com/dotcloud/docker/pkg/system"
 	"github.com/dotcloud/docker/utils"
-	"github.com/dotcloud/docker/vendor/src/code.google.com/p/go/src/pkg/archive/tar"
+	"archive/tar"
 	"io"
 	"io/ioutil"
 	"os"
diff --git a/archive/archive_test.go b/archive/archive_test.go
index 4126601..ca5a80c 100644
--- a/archive/archive_test.go
+++ b/archive/archive_test.go
@@ -3,7 +3,7 @@ package archive
 import (
 	"bytes"
 	"fmt"
-	"github.com/dotcloud/docker/vendor/src/code.google.com/p/go/src/pkg/archive/tar"
+	"archive/tar"
 	"io"
 	"io/ioutil"
 	"os"
diff --git a/archive/changes.go b/archive/changes.go
index 723e4a7..51baa22 100644
--- a/archive/changes.go
+++ b/archive/changes.go
@@ -5,7 +5,7 @@ import (
 	"fmt"
 	"github.com/dotcloud/docker/pkg/system"
 	"github.com/dotcloud/docker/utils"
-	"github.com/dotcloud/docker/vendor/src/code.google.com/p/go/src/pkg/archive/tar"
+	"archive/tar"
 	"io"
 	"os"
 	"path/filepath"
diff --git a/archive/diff.go b/archive/diff.go
index e20e4b1..a67b48a 100644
--- a/archive/diff.go
+++ b/archive/diff.go
@@ -2,7 +2,7 @@ package archive
 
 import (
 	"fmt"
-	"github.com/dotcloud/docker/vendor/src/code.google.com/p/go/src/pkg/archive/tar"
+	"archive/tar"
 	"io"
 	"io/ioutil"
 	"os"
diff --git a/archive/wrap.go b/archive/wrap.go
index 03ea508..0abe320 100644
--- a/archive/wrap.go
+++ b/archive/wrap.go
@@ -2,7 +2,7 @@ package archive
 
 import (
 	"bytes"
-	"github.com/dotcloud/docker/vendor/src/code.google.com/p/go/src/pkg/archive/tar"
+	"archive/tar"
 	"io/ioutil"
 )
 
diff --git a/graph/tags_unit_test.go b/graph/tags_unit_test.go
index 1777391..5cfe4c2 100644
--- a/graph/tags_unit_test.go
+++ b/graph/tags_unit_test.go
@@ -6,7 +6,7 @@ import (
 	"github.com/dotcloud/docker/runtime/graphdriver"
 	_ "github.com/dotcloud/docker/runtime/graphdriver/vfs" // import the vfs driver so it is used in the tests
 	"github.com/dotcloud/docker/utils"
-	"github.com/dotcloud/docker/vendor/src/code.google.com/p/go/src/pkg/archive/tar"
+	"archive/tar"
 	"io"
 	"os"
 	"path"
diff --git a/integration/api_test.go b/integration/api_test.go
index 26441a2..a313382 100644
--- a/integration/api_test.go
+++ b/integration/api_test.go
@@ -21,7 +21,7 @@ import (
 	"github.com/dotcloud/docker/runconfig"
 	"github.com/dotcloud/docker/runtime"
 	"github.com/dotcloud/docker/utils"
-	"github.com/dotcloud/docker/vendor/src/code.google.com/p/go/src/pkg/archive/tar"
+	"archive/tar"
 )
 
 func TestGetEvents(t *testing.T) {
diff --git a/integration/utils_test.go b/integration/utils_test.go
index 8ad6ccb..5b16f2a 100644
--- a/integration/utils_test.go
+++ b/integration/utils_test.go
@@ -3,7 +3,7 @@ package docker
 import (
 	"bytes"
 	"fmt"
-	"github.com/dotcloud/docker/vendor/src/code.google.com/p/go/src/pkg/archive/tar"
+	"archive/tar"
 	"io"
 	"io/ioutil"
 	"net/http"
diff --git a/utils/tarsum.go b/utils/tarsum.go
index 67e94aa..5132194 100644
--- a/utils/tarsum.go
+++ b/utils/tarsum.go
@@ -5,7 +5,7 @@ import (
 	"compress/gzip"
 	"crypto/sha256"
 	"encoding/hex"
-	"github.com/dotcloud/docker/vendor/src/code.google.com/p/go/src/pkg/archive/tar"
+	"archive/tar"
 	"hash"
 	"io"
 	"sort"
